tests/AiDotNet.Tensors.Benchmarks/BaselineRunners/py/run_direct_ptx_mixed_linear_competitors.py

```
@@ -0,0 +1,156 @@
+        shapes = SHAPES if args.rows == 1 else M16_SHAPES
+        for name, input_features, output_features in shapes:
+            x_shape = (input_features,) if args.rows == 1 else (args.rows, input_features)
+            x = (((torch.rand(x_shape, device=device) * 2.0 - 1.0) * 0.125)
+                 .to(torch.float16))
+            weights = (((torch.rand((output_features, input_features), device=device) * 2.0 - 1.0)
+                        * 0.0625).to(torch.float16))
+            # FP32 bias/output matches the direct kernel's public ABI. PyTorch
+            # promotes the FP16 dot result before this FP32 epilogue.
+            bias = ((torch.rand(output_features, device=device) * 2.0 - 1.0) * 0.0625)
+
+            def operation():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `operation`

(ANN202)